src/nvim/spell.c

```diff
@@ -2328,8 +2328,15 @@ static void spell_load_lang(char_u *lang)
   }
 
   if (r == FAIL) {
-    smsg(_("Warning: Cannot find word list \"%s.%s.spl\" or \"%s.ascii.spl\""),
-          lang, spell_enc(), lang);
+    if (starting) {
+      // Some startup file sets &spell, but the necessary files don't exist:
+      // try to prompt the user at VimEnter.
+      // Also, set spell again (downloading the files resets the option)
+      do_cmdline_cmd("au VimEnter * call spellfile#LoadFile(&spelllang) | set spell");
```

Comment:
It's because the function called is not defined, since it is defined in the spellfile plugin. No way around it, since this functionality depends on the runtime. The alternative is to bake it into nvim's codebase, but that means we would need an api to download files...